closures.js

```diff
@@ -128,3 +128,270 @@ function rollCall(names) {
 // rollCaller() // -> Should log 'Ruth'
 // rollCaller() // -> Should log 'Everyone accounted for'
 
+// CHALLENGE 9
+function cycleIterator(array) {
+  let count = 0;
+  const innerFunc = () => {
+    if (count < array.length) {
+      count++;
+    }
+    if (count >= array.length) {
+      count = 0;
+    }
+    return array[count];
+  };
+  return innerFunc;
+}
```

Comment:
It won't work as expected because the first output would be `Sat` as we are incrementing the counter on first iteration and it will return `array[1]`.

Reply:
function cycleIterator(array) {
  let counter = 0;
  let arrLength = array.length;
  
  return () => {
    return array[counter >= arrLength ? counter = 0 : counter++];
  }
}

Reply:
function cycleIterator(array) {
  let counter = 0;
  
  return () => {
    if( counter >= array.length) counter = 0;
    return array[counter++];
  }
}